Miscellaneous updates (#1640)
infiniflow/infinity: docs/references/pysdk_api_reference.md

### Examples

```python

> db_object=infinity_object.get_database("my_database")

Reviewer: db_object = infinity_object

## create_index

```python

> table_object.create_index(index_name, index_infos, conflict_type = ConflictType.Error)

Reviewer: index_infos => index_info

  - Digits (0-9)
  - "_" (underscore)

> #### index_infos: `IndexInfo()`, *Required*

Reviewer: index_infos => index_info

```python
# Select all columns
table_object.output(["*"]).to_pl()
```

```python
# Select columns "num" and "body"
table_object.output(["num", "body"]).to_df()
```

```python
# Select a system-generated column "_row_id"
table_object.output(["_row_id"]).to_result()
```

#### Perform aggregation or arithmetic operations on selected columns

```python
# Specify that the output should display the average value of all cells in column "c2"
table_object.output(["avg(c2)"]).to_result()
```

```python
# Select column "c1" and request all cells in this column to be displayed with their original values increased by 5
table_object.output(["c1+5"]).to_result()
```

```python
# Specify that the output should display the result of an arithmetic operation combining two aggregation functions
table_object.output(["min(c1) + max(c2)"]).to_result()
```

```python
# Specify that the output should display the row number of the current table
table_object.output(["count(*)"]).to_result()
```

```python
# Select column "num" and request all cells in this column to be displayed with their original values divided by 10

> table_object.output(["num/10"]).to_pl()

Reviewer: num / 5

## filter

```python

> table_obj.output(columns).filter(cond)

Reviewer: remove output

Reviewer: table_obj => table_object

```

Creates a filtering condition expression for the current table.

:::tip NOTE

> Call `filter(cond)` in a chain after calling `output(columns)` on the same table object.

Reviewer: Not after calling

### Examples

```python
table_object.output(["c1", "c2"]).filter("(-7 < c1 or 9 >= c1) and (c2 = 3)").to_pl()
```

```python

> table_object.output(["*"]).filter("c2 = 3").to_result()

Reviewer: to_pl()

:::

#### matching_text: `str`, *Required*

A non-empty text string to search for. You can use various search options within the matching text, including:

- Single terms: `"blooms"`
- OR multiple terms: `"Bloom filter"`
- Phrase search: `'"Bloom filter"'`
- AND multiple terms: "space efficient"
- Escaping reserved characters: "space\-efficient"
- Sloppy phrase search: "harmful chemical"~10
- Field-specific search: title:(quick OR brown) AND body:foobar

#### options_text: `str`, *Required*

A non-empty string specifying the following search options:

- `"topn"`: `str`, *Required*  
  Specifies the number of the most relevant rows to retrieve, e.g., `"topn=10"` to obtain the ten most relevant rows.
- `"operator"`: `str`, *Optional*
  - If not specified, the search follows Infinity's full-text search syntax, meaning that logical and arithmetic operators and escape characters will function as full-text search operators, such as:
    - `&&`, `+`, `||`, `!`, `NOT`, `AND`, `OR` `-`, `(`, `)`, `~`, `^`, `:`, `"`.
    - Escape characters like `\`, `\t`, and more.

>   - If specified, the above logical and arithmetic operators will be treated as part of the search text, and the corresponding operator will be interpolated into `matching_text`.

Reviewer 2: infinity
```suggestion
  - If specified, infinity's full-text search syntax will not be applied, and the corresponding operator will be used.
```

    - `"operator=OR"`/`"operator=or"`: Interpolates the `OR` operator between words in `matching_text` to create a new search text.

>     - `"operator=AND"`/`"operator=and"`: Interpolates the `AND` operator between words in `matching_text` to create a new search text. Useful for searching text including code numbers like `"A01-233:BC"`, resulting in `"(A01) AND (-233) AND (:BC)"`.

Reviewer 2: ```suggestion
    - `"operator=AND"`/`"operator=and"`: Interpolates the `AND` operator between words in `matching_text` to create a new search text. Useful for searching text including code numbers like `"A01-233:BC"`, resulting in `"A01" AND "-233" AND "BC"`.
```

:::danger NOTE  
For now, it is only possible to create a full-text index on multiple columns.
:::

### Parameters

#### fields: `str`, *Required*

A non-empty, comma-separated string of column names on which the full-text search will be performed.

:::tip NOTE
Ensure that you have created a full-text index on these columns before executing a full-text search; otherwise, an error will occur.
:::

#### matching_text: `str`, *Required*

A non-empty text string to search for. You can use various search options within the matching text, including:

- Single terms: `"blooms"`
- OR multiple terms: `"Bloom filter"`
- Phrase search: `'"Bloom filter"'`
- AND multiple terms: "space efficient"
- Escaping reserved characters: "space\-efficient"
- Sloppy phrase search: "harmful chemical"~10
- Field-specific search: title:(quick OR brown) AND body:foobar

#### options_text: `str`, *Required*

A non-empty string specifying the following search options:

- `"topn"`: `str`, *Required*  
  Specifies the number of the most relevant rows to retrieve, e.g., `"topn=10"` to obtain the ten most relevant rows.
- `"operator"`: `str`, *Optional*   

>   - If not specified, the search follows the default full-text search syntax, meaning that logical and arithmetic operators and escape characters will function as full-text search operators, such as:

Reviewer: ```suggestion
  - If not specified, the search follows Infinity's full-text search syntax, meaning that logical and arithmetic operators and escape characters will function as full-text search operators, such as:
```

>     - `&&`, `+`, `||`, `!`, `NOT`, `AND`, `OR` `-`, `,`, `~`, `^`, `:`, `""`.

Reviewer: `""` => `"`

    - Escape characters like `\`, `\t`, and more.
  - If specified, the above logical and arithmetic operators will be treated as part of the search text, and the corresponding operator will be interpolated into `matching_text`.
    - `"operator=OR"`/`"operator=or"`: Interpolates the `OR` operator between words in `matching_text` to create a new search text.

>     - `"operator=AND"`/`"operator=and"`: Interpolates the `AND` operator between words in `matching_text` to create a new search text. Useful for searching text including code numbers like `"A01-233:BC\显卡"`, resulting in `"(A01) AND (-233) AND (:BC) AND (显卡)"`.

Reviewer: Remove chinese word

## match

```python

> table_obj.match(fields, matching_text, distance_type, options_text)

Reviewer: one more options added.

Reviewer: "topn=5;operator=or" and "topn=5;operator=and"

```

Performs a full-text search on the specified field(s)/column(s) and returns the most relevant rows to the provided matching text.

:::danger NOTE  

> For now, it is only possible to create a full-text indext on multiple columns.

Reviewer: Not allow creating multiple columns index.

:::

### Parameters

#### fields: `str`, *Required*

A non-empty, comma-separated string of column names on which the full-text search will be performed.

:::tip NOTE
Ensure that you have created a full-text index on these columns before executing a full-text search; otherwise, an error will occur.

#### matching_text: `str`, *Required*

A non-empty text string to search for. You can use various search options within the matching text, including:

- Single terms: `"blooms"`
- OR multiple terms: `"Bloom filter"`
- Phrase search: `'"Bloom filter"'`
- AND multiple terms: "space efficient"
- Escaping reserved characters: "space\-efficient"

> - Sloppy phrase search: "harmful chemical"~10

Reviewer: Examples will be good.

Reviewer 3: [List of reserved characters](https://github.com/infiniflow/infinity/blob/67acc91b598626a4ba58039f78dd37c02f02e70d/src/storage/invertedindex/search/search_driver.cpp#L260):

                case ' ':
                case '+':
                case '-':
                case '=':
                case '&':
                case '|':
                case '!':
                case '(':
                case ')':
                case '{':
                case '}':
                case '[':
                case ']':
                case '^':
                case '"':
                case '~':
                case '*':
                case '?':
                case ':':
                case '\\':
                case '/':

- Field-specific search: title:(quick OR brown) AND body:foobar

#### options_text: `str`, *Required*

> Currently, only `topn` can be set in `options_text`, e.g., `"topn=10"` to retrieve the ten most relevant rows.

Reviewer: "topn=5;operator=or" and "topn=5;operator=and"

---

## to_result

```python
table_obj.to_result()
```

Returns the current table object as a tuple.

:::tip NOTE
Call `to_result()` in a chain after (not necessarily "immediately after") `output(columns)` on the same table object.
:::

:::caution NOTE

> We recommend calling `to_df()`, `to_pl()`, or `to_arrow()` to save your results.

Reviewer: format

:::

### Returns 

`tuple[dict[str, list[Any]], dict[str, Any]]`

## to_df

```python
table_obj.to_df()
```

Returns the current table object as a pandas DataFrame.

:::tip NOTE
Call `to_df()` in a chain after (not necessarily "immediately after") `output(columns)` on the same table object.
:::

### Examples

```python
# Save columns "c1" and C2" of the current table into a pandas DataFrame
res = table_obj.output(["c1", "c2"]).to_df()
```

### Returns

A `pandas.DataFrame` object.

## to_pl

```python
table_obj.to_pl()
```

> Returns the current table object as a Polas DataFrame.

Reviewer: Similar as above

:::tip NOTE
Call `to_pl()` in a chain after (not necessarily "immediately after") `output(columns)` on the same table object.
:::

### Returns

A `polas.DataFrame` object.

### Examples

```python
# Save a vector search result into a Polas DataFrame. 
res = table_obj.output(["*"])
               .knn("vec", [3.0, 2.8, 2.7, 3.1], "float", "ip", 10)
               .to_pl()
```

## to_arrow

```python
table_obj.to_arrow()
```

> Returns the current table object as an [Apache Arrow Table](https://arrow.apache.org/docs/python/generated/pyarrow.Table.html#pyarrow.Table).

Reviewer: Similar as above

Reviewer: Remove the link

:::note

> Call `to_df()` in a chain after (not necessarily "immediately after") `output(columns)` on the same table object.

Reviewer: to_arrow
